utilities/fault_injection_fs.cc

```diff
@@ -803,11 +803,15 @@ IOStatus FaultInjectionTestFS::InjectThreadSpecificReadError(
       *result = Slice();
       ctx->message += "inject empty result; ";
       ret_fault_injected = true;
-    } else if (!direct_io && Random::GetTLSInstance()->OneIn(7)) {
+    } else if (!direct_io && Random::GetTLSInstance()->OneIn(7) &&
+               scratch != nullptr && result->data() == scratch) {
```

Comment:
tangent: I hope one day mmap can use a different function that doesn't require providing an unused scratch buffer